crypto/crmf/crmf_lib.c

```diff
@@ -0,0 +1,751 @@
+/*
+ * Copyright 2007-2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright Nokia 2007-2018
+ * Copyright Siemens AG 2015-2018
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * CRMF implementation by Martin Peylo, Miikka Viljanen, and David von Oheimb.
+ */
+
+/* NAMING
+ * The 0 versions use the supplied structure pointer directly in the parent and
+ * it will be freed up when the parent is freed. In the above example crl would
+ * be freed but rev would not.
+ *
+ * The 1 functions use a copy of the supplied structure pointer (or in some
+ * cases increases its link count) in the parent and so both should be freed up.
+ */
+
+/*
+ * This file contains the functions that handle the individual items inside
```

Comment:
would it make sinse to move this comment before the prevoius?

Reply:
Yes, I just had the same thought; done.